File: ios/Classes/MoussaHlsPlayerPlugin.swift
import Flutter
import UIKit

public class MoussaHlsPlayerPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let factory = MoussaHlsNativeViewFactory(messenger: registrar.messenger())
    registrar.register(factory, withId: "moussa_hls_player/native_view")
  }
}

File: android/src/main/kotlin/com/moussait/moussa_hls_player/MoussaHlsPlayerPlugin.kt
package com.moussait.moussa_hls_player

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding

class MoussaHlsPlayerPlugin : FlutterPlugin {

  override fun onAttachedToEngine(binding: FlutterPluginBinding) {
    binding
      .platformViewRegistry
      .registerViewFactory(
        "moussa_hls_player/native_view",
        MoussaHlsNativeViewFactory(binding.binaryMessenger, binding.applicationContext)
      )
  }

  override fun onDetachedFromEngine(binding: FlutterPluginBinding) {
    // nothing
  }
}
